src/main/webapp/cluster-view/cluster-view.controller.js

```diff
@@ -1,26 +1,170 @@
 'use strict';
 
 angular.module('managementConsole')
-    .controller('ClusterViewCtrl', [
+  .controller('ClusterViewCtrl', [
     '$scope',
     '$stateParams',
     '$state',
+    '$q',
     'modelController',
-    function ($scope, $stateParams, $state, modelController) {
-            if (!modelController.isAuthenticated()) {
-                $state.go('/logout');
+    function ($scope, $stateParams, $state, $q, modelController) {
+      if (!modelController.isAuthenticated()) {
+        $state.go('/logout');
+      }
+      $scope.shared = {
+        currentCollection: 'caches'
+      };
+      $scope.clusters = modelController.getServer().getClusters();
+      $scope.currentCluster = modelController.getServer().getCluster($scope.clusters, $stateParams.clusterName);
+
+      $scope.$watch('currentCluster', function (currentCluster) {
+        if (currentCluster && currentCluster.name !== $stateParams.clusterName) {
+          $state.go('clusterView', {
+            'clusterName': currentCluster.name
+          });
+        }
+      });
+
+      $scope.isNodesView = function (){
+        return $scope.shared.currentCollection === 'nodes';
+      };
+
+      $scope.isCachesView = function (){
+        return $scope.shared.currentCollection === 'caches';
+      };
+
+      $scope.zoomedIn = false;
+
+      var zoomChanged = function () {
+        var slid = $("#zoom");
```

Comment:
Generally, there should be no DOM manipulation in angular controller: controller should contain only bussiness logic, basically only connecting other components. All DOM manipulation should be done in directives, and shared logic should be done in services. Here, we have DOM manipulation, so we should consider moving it to directive. Same applies to sliders below. I was not able to run the app yet, but will comment more on this when I get it running.